fluster/decoders/gstreamer.py

```diff
@@ -71,10 +72,56 @@ def __init__(self) -> None:
         if not gst_element_exists(self.sink):
             self.sink = 'filesink'
 
-    def gen_pipeline(self, input_filepath: str, output_filepath: str, output_format: OutputFormat) -> str:
+    def gen_pipeline(self, input_filepath: str, output_filepath: Optional[str], output_format: OutputFormat) -> str:
         '''Generate the GStreamer pipeline used to decode the test vector'''
         # pylint: disable=unused-argument
-        return PIPELINE_TPL.format(self.cmd, input_filepath, self.decoder_bin, self.caps, self.sink, output_filepath)
+        output = "location={}".format(output_filepath) if output_filepath else ""
+        return PIPELINE_TPL.format(self.cmd, input_filepath, self.decoder_bin, self.caps, self.sink, output)
+
+    def parse_md5sum(self, data: Tuple[str, str], verbose: bool) -> str:
+        '''Parse the MD5 sum out of commandline output'''
+        md5sum = "error"
+        for line in filter(None, data):
+            if verbose:
+                print(line, end='')
+            pattern = "conformance/checksum, checksum-type=(string)MD5, checksum=(string)"
+            sum_start = line.find(pattern)
+            if sum_start > 0:
+                sum_start += len(pattern)
+                sum_end = line[sum_start:].find(";")
+                if sum_end > 0:
+                    sum_end += sum_start
+                    md5sum = line[sum_start:sum_end]
+        return md5sum
```

Comment:
> Question 1: This is so that the output is complete in verbose mode.
Ah okay, I guess it is no big deal performance wise, but maybe you could do something like
```
  md5sum = line[sum_start:sum_end
+ if not verbose:
+     return md5sum
```

> Question 2: I have no idea how python regex works ;-P, if you think a regex performs better, please share a working diff please.

It won't perform any better as a regex is more expensive than a simple iteration. But it is shorter code, I think it is a matter of preference as the performance loss of the regex isn't bad either.
I just wanted to highlight a viable alternative but I think your approach is fine as well.

Reply:
Added the suggested if not verbose, kept the rest though.